src/factorizations.jl

```diff
@@ -305,20 +305,32 @@ _qr!(layout, axes, A, args...; kwds...) = error("Overload _qr!(::$(typeof(layout
 _lu(layout, axes, A; kwds...) = Base.invoke(lu, Tuple{AbstractMatrix{eltype(A)}}, A; kwds...)
 _lu(layout, axes, A, pivot::P; kwds...) where P = Base.invoke(lu, Tuple{AbstractMatrix{eltype(A)},P}, A, pivot; kwds...)
 _lu!(layout, axes, A, args...; kwds...) = error("Overload _lu!(::$(typeof(layout)), axes, A)")
-_cholesky(layout, axes, A, ::Val{false}=Val(false); check::Bool = true) = cholesky!(cholcopy(A); check = check)
-_cholesky(layout, axes, A, ::Val{true}; tol = 0.0, check::Bool = true) = cholesky!(cholcopy(A), Val(true); tol = tol, check = check)
+_cholesky(layout, axes, A, ::CNoPivot=CNoPivot(); check::Bool = true) = cholesky!(cholcopy(A); check = check)
+_cholesky(layout, axes, A, ::CRowMaximum; tol = 0.0, check::Bool = true) = cholesky!(cholcopy(A), CRowMaximum(); tol = tol, check = check)
 _factorize(layout, axes, A) = qr(A) # Default to QR
 
 
 _factorize(::AbstractStridedLayout, axes, A) = lu(A)
-function _lu!(::AbstractColumnMajor, axes, A::AbstractMatrix{T}, pivot::Union{NoPivot, RowMaximum} = RowMaximum();
-            check::Bool = true) where T<:BlasFloat
-    if pivot === NoPivot()
-        return generic_lufact!(A, pivot; check = check)
+if VERSION < v"1.8-"
```

Comment:
Wouldn't it make more sense to put the conditional in the function body? Optionally with a `@static`, but I don't think that is necessary for performance because `VERSION` is a `const` so Julia will optimize the conditional away.

Reply:
I slightly prefer my version because it doesn’t rely on the compiler optimising (which will be annoying to debug) but if you are confident please feel free to change